lib/node_modules/@stdlib/array/complex64/lib/main.js

```diff
@@ -1517,6 +1517,63 @@ setReadOnlyAccessor( Complex64Array.prototype, 'length', function get() {
 	return this._length;
 });
 
+/**
+* Returns a new array with each element being the result of the callback function.
+*
+* @name map
+* @memberof Complex64Array.prototype
+* @type {Function}
+* @param {Function} fcn - callback function
+* @param {*} [thisArg] - callback function execution context
+* @throws {TypeError} `this` must be a complex number array
+* @throws {TypeError} first argument must be a function
+* @returns {Complex64Array} complex number array
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float32' );
+* var realf = require( '@stdlib/complex/realf' );
+* var imagf = require( '@stdlib/complex/imagf' );
+*
+* function scale( v, i ) {
+*     return new Complex64( 2.0*realf( v ), 2.0*imagf( v ) );
+* }
+*
+* var arr = new Complex64Array( 3 );
+*
+* arr.set( [ 1.0, -1.0 ], 0 );
+* arr.set( [ 2.0, -2.0 ], 1 );
+* arr.set( [ 3.0, -3.0 ], 2 );
+*
+* var out = arr.map( scale );
+* // returns <Complex64Array>
+*
+* var z = out.get( 0 );
+* // returns <Complex64>
+*
+* var re = realf( z );
+* // returns 2.0
+*
+* var im = imagf( z );
+* // returns -2.0
+*/
+setReadOnly( Complex64Array.prototype, 'map', function map( fcn, thisArg ) {
+	var buf;
+	var out;
+	var i;
+	if ( !isComplexArray( this ) ) {
+		throw new TypeError( 'invalid invocation. `this` is not a complex number array.' );
+	}
+	if ( !isFunction( fcn ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a function. Value: `%s`.', fcn ) );
+	}
+	buf = this._buffer;
+	out = [];
+	for ( i = 0; i < this._length; i++ ) {
+		out.push( fcn.call( thisArg, getComplex64( buf, i ), i, this ) );
```

Comment:
@Jaysukh-409 This should be updated to support either returning a complex-like object, as you've currently implemented, or a two-element array of real and imaginary components. For a reference, see how `Complex64Array.from` is currently implemented.

Reply:
In particular, see https://github.com/stdlib-js/stdlib/blob/a1701983277c84238648e2aa5c0aa84dde12b0b5/lib/node_modules/%40stdlib/array/complex64/lib/main.js#L444 for an idea of how to handle, and similar to there, we want to throw an error if the callback does not return an acceptable value.